src/pages/ProfilesExplorerView/infrastructure/labels/LabelsDataSource.ts

```diff
@@ -73,6 +73,49 @@ export class LabelsDataSource extends RuntimeDataSource {
     };
   }
 
+  async fetchLabels(dataSourceUid: string, query: string, from: number, to: number, variableName?: string) {
+    labelsRepository.setApiClient(new LabelsApiClient({ dataSourceUid }));
+
+    try {
+      return await labelsRepository.listLabels({ query, from, to });
+    } catch (error) {
+      logger.error(error as Error, {
+        info: 'Error while loading Pyroscope label names!',
+        variableName: variableName || '',
+      });
+
+      throw error;
+    }
+  }
+
+  async fetchLabelValues(query: string, from: number, to: number, labelName: string, variableName?: string) {
+    let values;
+
+    try {
+      values = await labelsRepository.listLabelValues({ query, from, to, label: labelName });
+    } catch (error) {
+      logger.error(error as Error, {
+        info: 'Error while loading Pyroscope label values!',
+        variableName: variableName || '',
+      });
```

Comment:
Here we don't rethrow, we degrade gracefully so the UI can render something (hopefully) useful for the user.